### ISSUE TITLE
Exiting Message when trying to resume training on Colab

### ISSUE THREAD

**Issue by DerpyMan101** [NVlabs/stylegan2-ada-pytorch#101]
Whenever I try to resume training on colab pro, it stops with the message `Exiting...`.

**To Reproduce**
1. In `/content/drive/My Drive/colab-sg2-ada-pytorch/stylegan2-ada-pytorch` directory, run command `!python train.py --gpus=1 --cfg=$config --metrics=None --outdir=$output --data=$dataset_path --snap=$snapshot_count --resume=$resume_from --nkimg=$train_count`

2. See error (copy&paste full log, including exceptions and **stacktraces**).

`
Training options:
{
  "num_gpus": 1,
  "image_snapshot_ticks": 4,
  "network_snapshot_ticks": 4,
  "metrics": [],
  "random_seed": 0,
  "training_set_kwargs": {
    "class_name": "training.dataset.ImageFolderDataset",
    "path": "/content/drive/MyDrive/colab-sg2-ada-pytorch/stylegan2-ada-pytorch/datasets/output.zip",
    "use_labels": false,
    "max_size": 99119,
    "xflip": false,
    "resolution": 128
  },
  "data_loader_kwargs": {
    "pin_memory": true,
    "num_workers": 3,
    "prefetch_factor": 2
  },
  "G_kwargs": {
    "class_name": "training.networks.Generator",
    "z_dim": 512,
    "w_dim": 512,
    "mapping_kwargs": {
      "num_layers": 8
    },
    "synthesis_kwargs": {
      "channel_base": 32768,
      "channel_max": 512,
      "num_fp16_res": 4,
      "conv_clamp": 256
    }
  },
  "D_kwargs": {
    "class_name": "training.networks.Discriminator",
    "block_kwargs": {},
    "mapping_kwargs": {},
    "epilogue_kwargs": {
      "mbstd_group_size": 4
    },
    "channel_base": 32768,
    "channel_max": 512,
    "num_fp16_res": 4,
    "conv_clamp": 256
  },
  "G_opt_kwargs": {
    "class_name": "torch.optim.Adam",
    "lr": 0.002,
    "betas": [
      0,
      0.99
    ],
    "eps": 1e-08
  },
  "D_opt_kwargs": {
    "class_name": "torch.optim.Adam",
    "lr": 0.002,
    "betas": [
      0,
      0.99
    ],
    "eps": 1e-08
  },
  "loss_kwargs": {
    "class_name": "training.loss.StyleGAN2Loss",
    "r1_gamma": 10
  },
  "total_kimg": 25000,
  "batch_size": 4,
  "batch_gpu": 4,
  "ema_kimg": 10,
  "ema_rampup": null,
  "nimg": 25000000,
  "ada_target": 0.6,
  "augment_kwargs": {
    "class_name": "training.augment.AugmentPipe",
    "xflip": 1,
    "rotate90": 1,
    "xint": 1,
    "scale": 1,
    "rotate": 1,
    "aniso": 1,
    "xfrac": 1,
    "brightness": 1,
    "contrast": 1,
    "lumaflip": 1,
    "hue": 1,
    "saturation": 1
  },
  "resume_pkl": "/content/drive/MyDrive/colab-sg2-ada-pytorch/stylegan2-ada-pytorch/results2/00002-output-11gb-gpu-noaug-resumecustom/network-snapshot-000960.pkl",
  "ada_kimg": 100,
  "run_dir": "/content/drive/MyDrive/colab-sg2-ada-pytorch/stylegan2-ada-pytorch/results2/00006-output-11gb-gpu-resumecustom"
}

Output directory:   /content/drive/MyDrive/colab-sg2-ada-pytorch/stylegan2-ada-pytorch/results2/00006-output-11gb-gpu-resumecustom
Training data:      /content/drive/MyDrive/colab-sg2-ada-pytorch/stylegan2-ada-pytorch/datasets/output.zip
Training duration:  25000 kimg
Number of GPUs:     1
Number of images:   99119
Image resolution:   128
Conditional model:  False
Dataset x-flips:    False

Creating output directory...
Launching processes...
Loading training set...

Num images:  99119
Image shape: [3, 128, 128]
Label shape: [0]

Constructing networks...
Resuming from "/content/drive/MyDrive/colab-sg2-ada-pytorch/stylegan2-ada-pytorch/results2/00002-output-11gb-gpu-noaug-resumecustom/network-snapshot-000960.pkl"
Setting up PyTorch plugin "bias_act_plugin"... Done.
Setting up PyTorch plugin "upfirdn2d_plugin"... Done.

Generator             Parameters  Buffers  Output shape        Datatype
---                   ---         ---      ---                 ---     
mapping.fc0           262656      -        [4, 512]            float32 
mapping.fc1           262656      -        [4, 512]            float32 
mapping.fc2           262656      -        [4, 512]            float32 
mapping.fc3           262656      -        [4, 512]            float32 
mapping.fc4           262656      -        [4, 512]            float32 
mapping.fc5           262656      -        [4, 512]            float32 
mapping.fc6           262656      -        [4, 512]            float32 
mapping.fc7           262656      -        [4, 512]            float32 
mapping               -           512      [4, 12, 512]        float32 
synthesis.b4.conv1    2622465     32       [4, 512, 4, 4]      float32 
synthesis.b4.torgb    264195      -        [4, 3, 4, 4]        float32 
synthesis.b4:0        8192        16       [4, 512, 4, 4]      float32 
synthesis.b4:1        -           -        [4, 512, 4, 4]      float32 
synthesis.b8.conv0    2622465     80       [4, 512, 8, 8]      float32 
synthesis.b8.conv1    2622465     80       [4, 512, 8, 8]      float32 
synthesis.b8.torgb    264195      -        [4, 3, 8, 8]        float32 
synthesis.b8:0        -           16       [4, 512, 8, 8]      float32 
synthesis.b8:1        -           -        [4, 512, 8, 8]      float32 
synthesis.b16.conv0   2622465     272      [4, 512, 16, 16]    float16 
synthesis.b16.conv1   2622465     272      [4, 512, 16, 16]    float16 
synthesis.b16.torgb   264195      -        [4, 3, 16, 16]      float16 
synthesis.b16:0       -           16       [4, 512, 16, 16]    float16 
synthesis.b16:1       -           -        [4, 512, 16, 16]    float32 
synthesis.b32.conv0   2622465     1040     [4, 512, 32, 32]    float16 
synthesis.b32.conv1   2622465     1040     [4, 512, 32, 32]    float16 
synthesis.b32.torgb   264195      -        [4, 3, 32, 32]      float16 
synthesis.b32:0       -           16       [4, 512, 32, 32]    float16 
synthesis.b32:1       -           -        [4, 512, 32, 32]    float32 
synthesis.b64.conv0   2622465     4112     [4, 512, 64, 64]    float16 
synthesis.b64.conv1   2622465     4112     [4, 512, 64, 64]    float16 
synthesis.b64.torgb   264195      -        [4, 3, 64, 64]      float16 
synthesis.b64:0       -           16       [4, 512, 64, 64]    float16 
synthesis.b64:1       -           -        [4, 512, 64, 64]    float32 
synthesis.b128.conv0  1442561     16400    [4, 256, 128, 128]  float16 
synthesis.b128.conv1  721409      16400    [4, 256, 128, 128]  float16 
synthesis.b128.torgb  132099      -        [4, 3, 128, 128]    float16 
synthesis.b128:0      -           16       [4, 256, 128, 128]  float16 
synthesis.b128:1      -           -        [4, 256, 128, 128]  float32 
---                   ---         ---      ---                 ---     
Total                 29328669    44448    -                   -       

Discriminator  Parameters  Buffers  Output shape        Datatype
---            ---         ---      ---                 ---     
b128.fromrgb   1024        16       [4, 256, 128, 128]  float16 
b128.skip      131072      16       [4, 512, 64, 64]    float16 
b128.conv0     590080      16       [4, 256, 128, 128]  float16 
b128.conv1     1180160     16       [4, 512, 64, 64]    float16 
b128           -           16       [4, 512, 64, 64]    float16 
b64.skip       262144      16       [4, 512, 32, 32]    float16 
b64.conv0      2359808     16       [4, 512, 64, 64]    float16 
b64.conv1      2359808     16       [4, 512, 32, 32]    float16 
b64            -           16       [4, 512, 32, 32]    float16 
b32.skip       262144      16       [4, 512, 16, 16]    float16 
b32.conv0      2359808     16       [4, 512, 32, 32]    float16 
b32.conv1      2359808     16       [4, 512, 16, 16]    float16 
b32            -           16       [4, 512, 16, 16]    float16 
b16.skip       262144      16       [4, 512, 8, 8]      float16 
b16.conv0      2359808     16       [4, 512, 16, 16]    float16 
b16.conv1      2359808     16       [4, 512, 8, 8]      float16 
b16            -           16       [4, 512, 8, 8]      float16 
b8.skip        262144      16       [4, 512, 4, 4]      float32 
b8.conv0       2359808     16       [4, 512, 8, 8]      float32 
b8.conv1       2359808     16       [4, 512, 4, 4]      float32 
b8             -           16       [4, 512, 4, 4]      float32 
b4.mbstd       -           -        [4, 513, 4, 4]      float32 
b4.conv        2364416     16       [4, 512, 4, 4]      float32 
b4.fc          4194816     -        [4, 512]            float32 
b4.out         513         -        [4, 1]              float32 
---            ---         ---      ---                 ---     
Total          28389121    352      -                   -       

Setting up augmentation...
Distributing across 1 GPUs...
Setting up training phases...
Exporting sample images...
Initializing logs...
2021-05-03 13:19:23.544925: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
Training for 25000 kimg...

tick 0     kimg 25000.0  time 2m 00s       sec/tick 5.8     sec/kimg 1453.04 maintenance 114.4  cpumem 4.08   gpumem 9.76   augment 0.000

Exiting...`

**Expected behavior**
The training was supposed to resume from the specified checkpoint. 

![colab1](https://user-images.githubusercontent.com/14142982/116882811-7cd59080-abea-11eb-85a0-1c62a8d013e9.PNG)
![colab2](https://user-images.githubusercontent.com/14142982/116882810-7cd59080-abea-11eb-9253-da9a862f444d.PNG)
![colab3](https://user-images.githubusercontent.com/14142982/116882807-7c3cfa00-abea-11eb-9f5c-35773d2ceeb3.PNG)
![colab4](https://user-images.githubusercontent.com/14142982/116882812-7d6e2700-abea-11eb-83f0-299b18119688.PNG)

**Comment by ageroul** [comment]
Hi, 
I don't know if this answer your question but I see something that you might find useful:

- You want to train for 25000 kimg, right? So, you made some training and colab reached his GPU time limit and disconnected.
- So, (next day) you want to resume training from the last pkl file (from yesterday). You put the path of the pkl file in the "resume_from" parameter and you run training again.
- **Since you continue training from a previous state the number of kimg should be changed too. You started from "train_count=25000" but now (according to your last saved pkl file) you should change to "train_count=24040" (25000-960).
If you don't change the "train_count" you will always train for 25000 kimg.**

**Comment by ageroul** [comment]
> Whenever I try to resume training on colab pro, it stops with the message `Exiting...`.
> 

Hi,
I don't know if this answer your question but I see something that you might find useful:

You want to train for 25000 kimg, right? So, you made some training and colab reached his GPU time limit and disconnected.
So, (next day) you want to resume training from the last pkl file (from yesterday). You put the path of the pkl file in the "resume_from" parameter and you run training again.
Since you continue training from a previous state the number of kimg should be changed too. You started from "train_count=25000" but now (according to your last saved pkl file) you should change to "train_count=24040" (25000-960).
If you don't change the "train_count" you will always train for 25000 kimg.
